app/components/Editor.js

```diff
@@ -113,13 +115,36 @@ export default class Editor extends React.Component {
         this.redactor.insertion.insertHtml(replacement, 'start');
         this.redactor.selection.restoreMarkers();
       },
-      onReceiveNote: (_note) => {
-        // Empty
-      },
       clearUndoHistory: () => {
         // Called when switching notes to prevent history mixup.
         $R('#editor', 'module.buffer.clear');
       },
+      noteShouldBeRendered: async (note) => {
+        const shouldRenderNote = async () => {
+          const isUnsafeContent = this.checkIfUnsafeContent(note.content.text);
+          if (isUnsafeContent) {
+            const trustUnsafeContent = note.clientData['trustUnsafeContent'] ?? false;
+            if (!trustUnsafeContent) {
+              const result = await this.showUnsafeContentAlert();
+              if (result) {
+                this.setTrustUnsafeContent(note);
+              }
+              return result;
+            }
+          }
+          return true;
+        };
+
+        const renderNote = await shouldRenderNote();
+        if (!renderNote) {
+          $R('#editor', 'source.setCode', '');
```

Comment:
But here you're still having side-effects in a getter. A getter shouldn't change things, otherwise this will lead to lots of unwanted bugs and difficult to change behavior. What about doing all of this computation (shouldRenderNote, changing editor state) inside of setEditorText/setEditorRawText?

Reply:
`setEditorRawText` doesn't have the note item available (only raw text) so I can't get the client data (to check if the note has been trusted already).

Reply:
What about passing the note object then? Perhaps you can add a new delegate function `onNoteValuesChanged` and pass the note object. There you can determine whether to render the note and set the editor in a special state. Once the editor is in this special state, you can return and do nothing from setEditorText.

This way you wouldn't need shouldRenderNote, which I still feel is a bit ambiguous/esoteric.

Reply:
See updated source. `onNoteValueChange` is called just before `setEditorText` on editor-kit.